INSTALL.md

```diff
@@ -0,0 +1,176 @@
+# Installing VariKN
+
+## Table of contents
+
+- [Requirements for source installs](#requirements-for-source-installs)
+  - [How to install MacOsX requirements](#how-to-install-macosx-requirements)
+  - [How to install Linux requirements](#how-to-install-linux-requirements)  
+- [Install Python tools only](#install-python-tools-only)
+  - [Install from PYPI wheel](#install-from-pypi-wheel)
+  - [Install from PYPI source](#install-from-pypi-source)
+- [Full build](#full-build)
+  - [Build flags and debugging](#build-flags-and-debugging)
+  - [Builds fod IDE toolchains](#builds-fod-ide-toolchains)
+    - [XCode](#xcode)
+    - [Note on Eclipse CDT](#note-on-eclipse-cdt)
+  - [Installing python](#installing-python)
+  - [Testing](#testing)
+
+## Requirements for source installs
+
+If you are on Linux and only need python tools, you can jump to
+[Install from PYPI wheel](#install-from-pypi-wheel). For other install
+methods, you need a compiler, CMake, and Boost. For building the
+Python tools, SWIG is required.
+
+### How to install MacOsX requirements
+
+You need a compiler, for example in XCode Command Line
+Tools that can be installed with
+
+```sh
+xcode-select --install
+```
+
+Boost, CMake and SWIG can be installed for with Homebrew with the
+following command.
+
+```sh
+brew install boost cmake swig
+```
+
+### How to install Linux requirements
+
+Check the [Dockerfile](Dockerfile) for an example
+
+## Install Python tools only
+
+### Install from PYPI wheel
+
+In Linux you should be able to just run the following command to
+install.
+
+```sh
+python -m pip install varikn
+```
+
+If this for some reason fails, try the next section
+
+### Install from PYPI source
+
+You need to have the [required
+tools](#requirements-for-source-installs) to proceed.
+
+```sh
+python -m pip install varikn --no-binary :all:
+```
+
+## Full build
+
+You need to have the [required
+tools](#requirements-for-source-installs) to proceed.
+
+CMake is a makefile generator (like Autoconf). The build script is in
+CMakeLists.txt.  The convention for CMake is to build "out-of-source",
+where all the work is done in a separate directory from your source
+directory.  First, we configure the build (we only need to do this
+once):
+
+```sh
+mkdir build
+cd build
+cmake .. -DCMAKE_BUILD_TYPE=Release
+cmake --build .
+```
+
```

Comment:
```suggestion

Note that this will build the Python wrapper under `lib/python`, but
will not install it. See [Install from source](#install-from-source)
for simple Python installation. You can explicitly disable building
of the wrapper by passing `-DENABLE_PYTHON_WRAPPER=0` to
cmake.

```